test/functional/test_runner.py

```diff
@@ -30,6 +30,9 @@
 
 os.environ["REQUIRE_WALLET_TYPE_SET"] = "1"
 
+# Minimum amount of space to run the tests.
+MIN_FREE_SPACE = 1250 * 1024 * 1024
```

Comment:
@achow101 could you elaborate more on this
> Also, if --nocleanup is provided, we'd probably want to warn at the size where nothing is cleaned up.

Reply:
I have updated the PR as suggested:

> When a high enough parallelism is used, this is not enough.

>It may be better to scale this based on the number of jobs that will be run in parallel. Could put a lower bound of 1.1 GB for feature_block which is an outlier, then start increasing by 100 MB for each additional job that will be run.

Reply:
> @achow101 could you elaborate more on this
> 
> > Also, if --nocleanup is provided, we'd probably want to warn at the size where nothing is cleaned up.

`--nocleanup` requires about 12 GB, so if that option is passed, the check should be for that amount.

Reply:
updated the PR to include the check for minimum space required for `--nocleanup`